### PR TITLE
Commenting from queue

### PR DESCRIPTION
**Pull request by LW001**
As per Prime Dabbster's request, here's the ability to comment through the queue. Shout at me for errors.

https://trello.com/c/HfjzMHPX

### CODE REVIEW

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

**Comment by LW001** [comment]
[[admin-queue] Add comment reaction](https://trello.com/c/HfjzMHPX/65-admin-queue-add-comment-reaction)

File: Commands/Commands/queue.js [hunk 1]
```diff
@@ -484,8 +492,101 @@ commands.registerVote = {
...
+                          cause: 'email_search',
+                          message: (e.message !== undefined) ? e.message : JSON.stringify(e)
+                        }, e)
+                        msg.reply('an error occured, please try again later.').then(mailerr => {
+                          setTimeout(() => bot.Messages.delete(mailerr), 2500)
+                        })
+                      }
+                    })
+                  }
+                  }
```
Comment (Dougley) [inline Commands/Commands/queue.js]: Format please

File: Commands/Commands/queue.js [hunk 2]
```diff
@@ -484,8 +492,101 @@ commands.registerVote = {
...
+                              {
+                                name: `${entities.decode(data.comment.creator.name)} commented on this:`,
+                                value: entities.decode(data.comment.text),
+                                inline: false
+                              }
+                            ]
+                          }).then(deletreply => {
+                            setTimeout(() => bot.Messages.delete(deletreply), 5000)
+                          })
+                          cBack({
```
Comment (Dougley) [inline Commands/Commands/queue.js]: Don't callback, this is only for actual commands

File: Commands/Commands/queue.js [hunk 3]
```diff
@@ -484,8 +492,101 @@ commands.registerVote = {
...
+                      uvClient.v1.loginAs(email).then(c => {
+                        let content = q
+                        let id = ${doc.UvId}
+                        c.post(`forums/${config.uservoice.forumId}/suggestions/${id}/comments.json`, {
+                          comment: {
+                            text: content
+                          }
+                        }).then(data => {
+                          msg.reply('your comment was added.', false, {
+                            title: entities.decode(data.comment.suggestion.title),
```
Comment (Dougley) [inline Commands/Commands/queue.js]: Don't embed, this will only cause more clutter

File: Commands/Commands/queue.js [hunk 4]
```diff
@@ -526,6 +627,100 @@ commands.registerVote = {
...
+                        cause: 'email_search',
+                        message: (e.message !== undefined) ? e.message : JSON.stringify(e)
+                      }, e)
+                      msg.reply('an error occured, please try again later.').then(mailerr => {
+                        setTimeout(() => bot.Messages.delete(mailerr), 2500)
+                      })
+                    }
+                  })
+                }
+                }
```
Comment (Dougley) [inline Commands/Commands/queue.js]: Format please

File: Commands/Commands/queue.js [hunk 5]
```diff
@@ -526,6 +627,100 @@ commands.registerVote = {
...
+                  getMail(uvClient, msg.author.id).then(email => {
+                    uvClient.v1.loginAs(email).then(c => {
+                      let content = q
+                      let id = ${doc.UvId}
+                      c.post(`forums/${config.uservoice.forumId}/suggestions/${id}/comments.json`, {
+                        comment: {
+                          text: content
+                        }
+                      }).then(data => {
+                        msg.reply('your comment was added.', false, {
```
Comment (Dougley) [inline Commands/Commands/queue.js]: Don't embed

Comment (SteamingMutt) [inline Commands/Commands/queue.js]: Don't embed

File: Commands/Commands/queue.js [hunk 6]
```diff
@@ -526,6 +627,100 @@ commands.registerVote = {
...
+                            {
+                              name: `${entities.decode(data.comment.creator.name)} commented on this:`,
+                              value: entities.decode(data.comment.text),
+                              inline: false
+                            }
+                          ]
+                        }).then(deletreply => {
+                          setTimeout(() => bot.Messages.delete(deletreply), 5000)
+                        })
+                        cBack({
```
Comment (Dougley) [inline Commands/Commands/queue.js]: Don't callback

File: Commands/Commands/queue.js [hunk 7]
```diff
@@ -617,4 +812,23 @@ function wait(bot, msg) {
...
+    bot.Dispatcher.on('MESSAGE_CREATE', function doStuff(c) {
+      var time = setTimeout(() => {
+        resolve(null)
+        bot.Dispatcher.removeListener('MESSAGE_CREATE', doStuff)
+      }, config.timeouts.queueComment) // We won't wait forever for the person to anwser
+      if (c.message.channel.id !== msg.channel.id) return
+      if (c.message.author.id !== msg.author.id) return
+      if (c.message.content === 'x') resolve(false)
+      else {
+        resolve(c)
```
Comment (Dougley) [inline Commands/Commands/queue.js]: Resolve with c.message.content

File: Commands/Commands/queue.js [hunk 8]
```diff
@@ -526,6 +627,100 @@ commands.registerVote = {
...
+                  })
+                }
+                if (q === false) {
+                  msg.reply('The operation has been cancelled.').then(msgdelet => {
+                    setTimeout(() => bot.Messages.deleteMessages([aacmsg.id, msgdelet.id, msg.id], bot.Channels.find(c => c.name === 'admin-queue').id), 2500)
+                  })
+                }
+                else {
+                  bot.Messages.deleteMessages([aacmsg.id, msg.id], bot.Channels.find(c => c.name === 'admin-queue').id)
+                  fn: function (bot, msg, suffix, uvClient, cBack) {
```
Comment (Dougley) [inline Commands/Commands/queue.js]: Nice copy-paste, fn is specific for commands

File: Commands/Commands/queue.js [hunk 9]
```diff
@@ -484,8 +492,101 @@ commands.registerVote = {
...
+                    })
+                  }
+                  if (q === false) {
+                    msg.reply('The operation has been cancelled.').then(msgdelet => {
+                      setTimeout(() => bot.Messages.deleteMessages([aacmsg.id, msgdelet.id, msg.id], bot.Channels.find(c => c.name === 'admin-queue').id), 2500)
+                    })
+                  }
+                  else {
+                    bot.Messages.deleteMessages([aacmsg.id, msg.id], bot.Channels.find(c => c.name === 'admin-queue').id)
+                    fn: function (bot, msg, suffix, uvClient, cBack) {
```
Comment (Dougley) [inline Commands/Commands/queue.js]: Nice copy-paste, fn is specific for commands

File: Commands/Commands/queue.js [hunk 10]
```diff
@@ -484,8 +492,101 @@ commands.registerVote = {
...
+                      setTimeout(() => bot.Messages.deleteMessages([aacmsg.id, msgdelet.id, msg.id], bot.Channels.find(c => c.name === 'admin-queue').id), 2500)
+                    })
+                  }
+                  if (q === false) {
+                    msg.reply('The operation has been cancelled.').then(msgdelet => {
+                      setTimeout(() => bot.Messages.deleteMessages([aacmsg.id, msgdelet.id, msg.id], bot.Channels.find(c => c.name === 'admin-queue').id), 2500)
+                    })
+                  }
+                  else {
+                    bot.Messages.deleteMessages([aacmsg.id, msg.id], bot.Channels.find(c => c.name === 'admin-queue').id)
```
Comment (Dougley) [inline Commands/Commands/queue.js:509]: Why are you deleting the report?

File: Commands/Commands/queue.js [hunk 11]
```diff
@@ -484,8 +492,101 @@ commands.registerVote = {
             deleteFromUV(doc.UvId, uv, bot)
             r.db('DFB').table('queue').get(doc.id).delete().run().catch(bugsnag.nofify)
           }
-          break
-        }
+            else if (reaction.id ==='322990935212556291') {
+              bot.Channels.find(c => c.name === 'admin-queue').sendMessage(`Add a comment to ${doc.embed.title} by typing it into this channel, or type x to cancel.`).then(aacmsg => {
+                waitCommenting(bot, msg).then((q) => {
+                  if (q === null) {
+                    msg.reply('You took too long to comment, the operation has been cancelled.').then(msgdelet => {
```
Comment (Dougley) [inline Commands/Commands/queue.js]: msg is the bots message, send a regular message

File: Commands/Commands/queue.js [hunk 12]
```diff
@@ -484,8 +492,101 @@ commands.registerVote = {
...
+                            result: 'A new comment was created'
+                          })
+                        }).catch(e => {
+                          if (e.statusCode === 404) {
+                            msg.reply('unable to find a suggestion using your query.').then(err => {
+                              setTimeout(() => bot.Messages.delete(err), 2500)
+                            })
+                          } else {
+                            logger.log(bot, {
+                              cause: 'comment_add',
```
Comment (Dougley) [inline Commands/Commands/queue.js]: This needs its own case

File: Commands/Commands/queue.js [hunk 13]
```diff
@@ -526,6 +627,100 @@ commands.registerVote = {
...
+                          result: 'A new comment was created'
+                        })
+                      }).catch(e => {
+                        if (e.statusCode === 404) {
+                          msg.reply('unable to find a suggestion using your query.').then(err => {
+                            setTimeout(() => bot.Messages.delete(err), 2500)
+                          })
+                        } else {
+                          logger.log(bot, {
+                            cause: 'comment_add',
```
Comment (Dougley) [inline Commands/Commands/queue.js]: Own case required

**Review by Dougley — CHANGES_REQUESTED** [review]

**Comment by LW001** [comment]
Kinda won't be able to recover that repo, http://imgur.com/a/9CiLZ

Closing PR and reopening it later today, sorry for the inconvenience.

**Pull request closed without merging**